### PR TITLE
Additional validations for mcm related settings

### PR DESCRIPTION
**Pull request by ibrokethecloud**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

With the introduction of `rancher-manager-support` integration in harvester, additional validations are needed to ensure that some settings cannot be disabled / enabled to avoid corruption of underlying rancher managed resources.

**Solution:**

**Related Issue:**
https://github.com/harvester/harvester/issues/4010

**Test plan:**

### CODE REVIEW

File: pkg/webhook/resources/setting/validator.go [hunk 1]
```diff
@@ -703,6 +721,14 @@ func (v *settingValidator) validateUpdateRancherManagerSupport(oldSetting *v1bet
 		return werror.NewInvalidError("cannot enable Rancher Manager support. The multi-cluster-management feature is not enabled", "value")
 	}
 
+	clusterRegistrationURLValue, err := v.settingCache.Get(settings.ClusterRegistrationURL.Name)
+	if err != nil {
+		werror.NewInternalError(err.Error())
```
Comment (connorkuehl) [inline pkg/webhook/resources/setting/validator.go]:
```suggestion
		return werror.NewInternalError(err.Error())
```

File: pkg/webhook/resources/setting/validator.go [hunk 2]
```diff
@@ -703,6 +721,14 @@ func (v *settingValidator) validateUpdateRancherManagerSupport(oldSetting *v1bet
...
+	}
+
+	if clusterRegistrationURLValue.Value != "" {
+		werror.NewMethodNotAllowed(fmt.Sprintf("unable to enable %s, as cluster is already managed by another rancher", settings.RancherManagerSupport.Name))
```
Comment (connorkuehl) [inline pkg/webhook/resources/setting/validator.go]:
```suggestion
		return werror.NewMethodNotAllowed(fmt.Sprintf("unable to enable %s, as cluster is already managed by another rancher", settings.RancherManagerSupport.Name))
```

File: pkg/webhook/resources/setting/validator.go [hunk 3]
```diff
@@ -726,3 +752,38 @@ func validateDefaultVMTerminationGracePeriodSeconds(setting *v1beta1.Setting) er
...
+}
+
+func (v *settingValidator) validateClusterRegistrationPossible(oldSetting, newSetting *v1beta1.Setting) error {
+	if newSetting.Value == "" {
+		return nil
+	}
+
+	rancherManagementSupportSetting, err := v.settingCache.Get(settings.RancherManagerSupportSettingName)
+	if err != nil {
+		werror.NewInternalError(err.Error())
```
Comment (connorkuehl) [inline pkg/webhook/resources/setting/validator.go]:
```suggestion
		return werror.NewInternalError(err.Error())
```

**Review by connorkuehl — APPROVED** [review]
Just a couple of minor fixups, but otherwise LGTM. The tests added useful context for better understanding the validations:+1:

**Comment by ibrokethecloud** [comment]
thanks for spotting that

**Comment by ibrokethecloud** [comment]
Not needed any longer. We will be proceeding with the experimental vcluster addon, which mitigates requirements for additional checks on the local cluster.

**Pull request closed without merging**